### PR TITLE
FUSETOOLS-2150 - fix broken collapse/expand

### PR DESCRIPTION
**Pull request by lhein**

### CODE REVIEW

**Comment by lhein** [comment]
work in progress...do not merge!

**Comment by lhein** [comment]
I will create a new jira for the persistence of the collapse state.

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateFigureFeature.java [hunk 1]
```diff
@@ -156,6 +157,11 @@ public boolean canCreate(ICreateContext context) {
 		
 		AbstractCamelModelElement containerNode = containerBO instanceof AbstractCamelModelElement ? (AbstractCamelModelElement)containerBO : null;
 	
+		if (CollapseFeature.isCollapsed(getFeatureProvider(), containerNode)) {
+    		// we don't allow drop on a collapsed figure
+    		return false;
+    	}
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateFigureFeature.java]: indentation issue

File: editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/CollapseFeature.java [hunk 2]
```diff
@@ -151,16 +165,23 @@ public void collapseShape(PictogramElement pe) {
...
+	 * updates the border color to reflect the collapse state of the figure
+	 * 
+	 * @param cs
+	 * @param collapsed
+	 */
+	public void updateBorderStyle(ContainerShape cs, boolean collapsed) {
+		IGaService gaService = Graphiti.getGaService();
+		Color col = collapsed ? gaService.manageColor(getDiagram(), StyleUtil.CONTAINER_FIGURE_COLLAPSED_BORDER_COLOR) : gaService.manageColor(getDiagram(), StyleUtil.CONTAINER_FIGURE_BORDER_COLOR); 
+		GraphicsAlgorithm ga = cs.getGraphicsAlgorithm();
+		ga.setLineStyle(collapsed ? LineStyle.DOT : LineStyle.SOLID);
```
Comment (apupier) [inline editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/CollapseFeature.java]: I prefer with LineStyle.DASH

image

than LineStyle.DOT

image

What do you think?
DOT are not visible enough for me

**Pull request merged**